usermanagement/src/main/kotlin/com/story/usermanagement/storage/UserRepositiry.kt

```diff
@@ -0,0 +1,64 @@
+package com.story.usermanagement.storage
+
+import org.springframework.stereotype.Component
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient
+import software.amazon.awssdk.enhanced.dynamodb.TableSchema
+import software.amazon.awssdk.enhanced.dynamodb.model.PageIterable
+import software.amazon.awssdk.enhanced.dynamodb.model.QueryConditional
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue
+import software.amazon.awssdk.services.dynamodb.model.DeleteItemRequest
+
+interface UserRepository {
+    fun findByEmail(emailAddress: String): UserDynamoDto?
+    fun save(dto: UserDynamoDto)
+    fun delete(id: Long)
+}
+
+@Component
+class UserRepositoryImpl(
+    private val dynamoDbClient: DynamoDbClient,
+    private val dynamoDbEnhancedClient: DynamoDbEnhancedClient,
+    dynamoTableManager: DynamoTableManager
+) : UserRepository {
+
+    private val qualifiedUserTableName = dynamoTableManager.getTableName(UserDynamoDto::class)
+
+    init {
+        dynamoTableManager.getTable(UserDynamoDto::class)
+            ?:dynamoTableManager.createTable(UserDynamoDto::class)
```

Comment:
wait, is this going to create a table on the fly ? 
I suggest moving this part to application boot instead. 
you might want to register all the table's on application load and make sure it's created before application starts.

Reply:
Repository Layer usually shouldn't have any concern if the table is created or not, this shouldn't be handled here. IMO.
Let me know your thoughts.